src/main/java/bio/terra/service/filedata/azure/blobstore/AzureBlobStorePdao.java

```diff
@@ -112,13 +103,26 @@ public FSFileInfo copyFile(
         .bucketResourceId(storageAccountResource.getResourceId().toString());
   }
 
+  public BlobContainerClientFactory buildSourceClientFactory(UUID tenantId, String blobUrl) {
+    BlobUrlParts blobUrlParts = BlobUrlParts.parse(blobUrl);
+    if (isSignedUrl(blobUrlParts)) {
+      return getSourceClientFactory(blobUrl);
+    } else {
+      // Use application level authentication
+      return getSourceClientFactory(
+          blobUrlParts.getAccountName(),
+          resourceConfiguration.getAppToken(tenantId),
+          blobUrlParts.getBlobContainerName());
+    }
+  }
+
   public boolean deleteDataFileById(
```

Comment:
I think there is a refactoring opportunity here. You can make the  `BlobCrl.deleteBlob.deleteBlob` method return `true` if delete is successful or return `false` and log the exception if the blob is not present.

Reply:
Thanks for taking a look at these changes - But, I think these refactoring suggestions are out of scope for this ticket.

Reply:
I created a Jira for this (DR-2030)